src/Command/CreateCoreArchCommand.php

```diff
@@ -0,0 +1,231 @@
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Centreon\Command;
+
+use Centreon\Command\Model\ModelTemplate\ModelTemplate;
+use Centreon\Command\Service\{
+    CreateCoreArchCommandService,
+    CreateCoreCommandArchCommandService,
+    CreateCoreQueryArchCommandService
+};
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CreateCoreArchCommand extends Command
+{
+    public const COMMAND_NAME = 'centreon:create-core-arch';
+    public const COMMAND_CREATE = 'Create';
+    public const COMMAND_UPDATE = 'Update';
+    public const COMMAND_DELETE = 'Delete';
+    public const COMMAND_FIND = 'Find';
+    public const COMMAND_ACTION = [
+        self::COMMAND_CREATE,
+        self::COMMAND_UPDATE,
+        self::COMMAND_DELETE,
+        self::COMMAND_FIND,
+    ];
+    public const READ_REPOSITORY_TYPE = 'Read';
+    public const WRITE_REPOSITORY_TYPE = 'Write';
+    public const COMMAND_USECASES = [self::COMMAND_CREATE, self::COMMAND_UPDATE, self::COMMAND_DELETE];
+    public const QUERY_USECASES = [self::COMMAND_FIND];
+
+    private string $useCaseType;
+
+    private ModelTemplate $modelTemplate;
+
+    /**
+     * @param CreateCoreArchCommandService $commandService
+     * @param CreateCoreQueryArchCommandService $queryArchCommandService
+     * @param CreateCoreCommandArchCommandService $commandArchCommandService
+     */
+    public function __construct(
+        private CreateCoreArchCommandService $commandService,
+        private CreateCoreQueryArchCommandService $queryArchCommandService,
+        private CreateCoreCommandArchCommandService $commandArchCommandService
+    ) {
+        parent::__construct();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function configure(): void
+    {
+        $this
+            ->setName(self::COMMAND_NAME)
+            ->setDescription('Create architecture for a useCase')
+            ->setHelp('This command allows you to create classes for a useCase');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function interact(InputInterface $input, OutputInterface $output): void
+    {
+        $output->writeln('');
+        $output->writeln('You are going to create a use case architecture.');
+        $output->writeln("Let's answer few questions first !");
+        $output->writeln('');
+
+        $questionHelper = $this->getHelper('question');
```

Comment:
At least if not checked more hardly.
```suggestion
        /** @var QuestionHelper $questionHelper */
        $questionHelper = $this->getHelper('question');
```